### ISSUE TITLE
[CC]Client freeze with casting stand

### ISSUE THREAD

**Issue by wormzjl** [ReikaKalseki/Reika_Mods_Issues#1510]
Caused by putting a thaumcraft axe of the core on the CC casing stand, but could be caused by many other things too

Log:
https://pastebin.com/4FVgp6S7

Seems to be caused by
https://github.com/ReikaKalseki/DragonAPI/blob/master/Libraries/IO/ReikaTextureHelper.java#L206

The vanilla texture file is no longer there?

**Comment by ReikaKalseki** [comment]
Your crash is coming from the BC engine TESR, and mentions neither my code nor ThaumCraft.

**Comment by wormzjl** [comment]
This can be replicated without BC and fastcraft, and same thing occurs if you try to put a knowledge fragment on a casting stand
And it's a freeze not a crash

**Comment by wormzjl** [comment]
Got this in the thread dump, seems a bit weird for me, issue updated
```
at Reika.ChromatiCraft.TileEntity.Recipe.TileEntityItemStand.spawnItemParticles(TileEntityItemStand.java:93)
at Reika.ChromatiCraft.TileEntity.Recipe.TileEntityItemStand.updateEntity(TileEntityItemStand.java:80)
at Reika.DragonAPI.Base.TileEntityBase.func_145845_h(TileEntityBase.java:498)
```
```

"Client thread" #1 prio=5 os_prio=0 tid=0x0000000002038800 nid=0x41d4 runnable [0x00000000024d9000]
   java.lang.Thread.State: RUNNABLE
        at net.minecraft.item.ItemStack.func_77960_j(ItemStack.java:217)
        at net.minecraft.item.crafting.FurnaceRecipes.func_151397_a(FurnaceRecipes.java:100)
        at net.minecraft.item.crafting.FurnaceRecipes.func_151395_a(FurnaceRecipes.java:93)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaSmelting(ItemElementCalculator.java:258)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:550)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:617)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaSmelting(ItemElementCalculator.java:260)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:550)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:617)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaSmelting(ItemElementCalculator.java:260)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:550)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaSmelting(ItemElementCalculator.java:260)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:550)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:626)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getTagForItemOrList(ItemElementCalculator.java:636)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getIRecipeTotal(ItemElementCalculator.java:609)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromVanillaCrafting(ItemElementCalculator.java:246)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:549)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getFromThaumCraft(ItemElementCalculator.java:409)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.calculateTag(ItemElementCalculator.java:558)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:206)
        at Reika.ChromatiCraft.Magic.ItemElementCalculator.getValueForItem(ItemElementCalculator.java:191)
        at Reika.ChromatiCraft.TileEntity.Recipe.TileEntityItemStand.spawnItemParticles(TileEntityItemStand.java:93)
        at Reika.ChromatiCraft.TileEntity.Recipe.TileEntityItemStand.updateEntity(TileEntityItemStand.java:80)
        at Reika.DragonAPI.Base.TileEntityBase.func_145845_h(TileEntityBase.java:498)
        at net.minecraft.world.World.func_72939_s(World.java:1939)
        at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:2006)
        at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:973)
        at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
        at net.minecraft.client.main.Main.main(SourceFile:148)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236)
        at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297)
        at org.multimc.EntryPoint.listen(EntryPoint.java:162)
        at org.multimc.EntryPoint.main(EntryPoint.java:53)

   Locked ownable synchronizers:
        - None

```

Full thread dump:
https://pastebin.com/FrRhfYsN

**Comment by ReikaKalseki** [comment]
This is a different issue from your first log.

**Comment by ReikaKalseki** [comment]
Not happening for me:
http://i.imgur.com/8ovUZLV.png
![](http://i.imgur.com/8ovUZLV.png)

Enchanted:
http://i.imgur.com/iR1CCNj.png
![](http://i.imgur.com/iR1CCNj.png)

**Comment by ReikaKalseki** [comment]
What ChromatiCraft version are you using?

**Comment by wormzjl** [comment]
V17c

**Comment by ReikaKalseki** [comment]
Then I have no explanation.

**Comment by wormzjl** [comment]
Weird...might be another Gregtech inteference.
Is it possible to add some limit to this particle calculation? Like, maximum depth?

**Comment by ReikaKalseki** [comment]
> another Gregtech inteference.

It has been known to cause this (#668, #1076, #1444).

And there *is* a depth limit.

**Comment by wormzjl** [comment]
I see that #668 is closed right now, do you have any plan to dig further into this issue?

**Comment by ReikaKalseki** [comment]
No, because there is nothing more I can do, and I have only a very vague idea of what is happening to begin with.

**Comment by Barhandar** [comment]
This is caused by interaction with GregTech, but NEI is unrelated. It is reproducible with just ChromatiCraft, GregTech 5 Unofficial, and their dependencies - DragonAPI and IC2. The versions are V18b for your mods, GT5U 5.09.30, and IC2E 2.2.827. The client freezes for an amount of time, no crashlog is generated, no error message in the regular log is generated, integrated server continues to work. 

Replication: add the aforementioned mods to new instance with latest Forge, start up, give yourself casting stand and Eye of Ender, place the Eye in the casting stand. The client will freeze while the server will not (there will be IDmap saving messages in the log from FML). After an amount of time that scales with instance complexity (for my PC, about two minutes for testing instance, unknown but longer than 20 minutes for my playing instance) the client will unfreeze and placing Eye of Ender will no longer freeze the instance until it is restarted completely. Placing a GT item afterwards (U238 screw) caused a hitch, afterwards no placed item caused any freezes or delays.  
On instance reload, loading the world with Eye of Ender in the casting stand causes the freeze as soon as it can run, so if alt-tabbed before it loads, when the ESC menu is quit.

I don't know what the code is doing when item is placed in casting stand, but could it please _not_. Alternately, the issue might be alleviated or altogether fixed by solutions to #668, if the casting stand triggers the same lookup.   
Also, can't that be made to not hang the entire thing by threading it?
